### PR TITLE
Extract bootstrap_servers from kafka_config

### PR DESCRIPTION
**Pull request by kumaranvpl**

### CODE REVIEW

File: fastkafka/_application/app.py [hunk 1]
```diff
@@ -390,6 +390,7 @@ def __init__(
         contact: Optional[Dict[str, str]] = None,
         kafka_brokers: Optional[Dict[str, Any]] = None,
         root_path: Optional[Union[Path, str]] = None,
+        bootstrap_servers: Optional[Union[str, List[str]]] = None,
```
Comment (davorrunje) [inline fastkafka/_application/app.py:393]: remove this from the constructor

File: fastkafka/_application/app.py [hunk 2]
```diff
@@ -676,7 +677,9 @@ def __init__(
         (self._asyncapi_path / "spec").mkdir(exist_ok=True, parents=True)
 
         # this is used as default parameters for creating AIOProducer and AIOConsumer objects
-        self._kafka_config = _get_kafka_config(**kwargs)
+        self._kafka_config = _get_kafka_config(
+            bootstrap_servers=bootstrap_servers, **kwargs
```
Comment (davorrunje) [inline fastkafka/_application/app.py:681]: remove this

File: fastkafka/_application/app.py [hunk 3]
```diff
@@ -716,6 +719,19 @@ def is_started(self) -> bool:
     def set_bootstrap_servers(self, bootstrap_servers: str) -> None:
```
Comment (davorrunje) [inline fastkafka/_application/app.py:719]: this should be private method starting with _

File: fastkafka/_application/app.py [hunk 4]
```diff
@@ -716,6 +719,19 @@ def is_started(self) -> bool:
...
         self._kafka_config["bootstrap_servers"] = bootstrap_servers
 
+    def set_kafka_broker(self, kafka_broker: str) -> None:
```
Comment (davorrunje) [inline fastkafka/_application/app.py:722]: kafka_broker_name

File: fastkafka/_application/app.py [hunk 5]
```diff
@@ -716,6 +719,19 @@ def is_started(self) -> bool:
...
+            )
+
+        broker_to_use = self._kafka_brokers.brokers[kafka_broker]
+        bootstrap_servers = f"{broker_to_use.url}:{broker_to_use.port}"
+        logger.info(
+            f"set_kafka_broker() : Setting bootstrap_servers value to '{bootstrap_servers}'"
+        )
+        self._kafka_config["bootstrap_servers"] = bootstrap_servers
+
     async def __aenter__(self) -> "FastKafka":
```
Comment (davorrunje) [inline fastkafka/_application/app.py:735]: optional parameter kafka_broker_name: Optional[str] = None

if set, remember the old value in the instance variable. Set it back to the old value in __aexit__

File: fastkafka/_cli.py [hunk 6]
```diff
@@ -33,15 +33,22 @@ def run(
         ...,
         help="input in the form of 'path:app', where **path** is the path to a python file and **app** is an object of type **FastKafka**.",
     ),
+    kafka_broker: Optional[str] = typer.Option(
+        None, help="broker to use from kafka_brokers"
```
Comment (davorrunje) [inline fastkafka/_cli.py:37]: kafka_broker, one of the keys of the kafka_brokers dictionary passed in the constructor of FastaKafka class.

File: fastkafka/_cli.py [hunk 7]
```diff
@@ -33,15 +33,22 @@ def run(
         ...,
         help="input in the form of 'path:app', where **path** is the path to a python file and **app** is an object of type **FastKafka**.",
     ),
+    kafka_broker: Optional[str] = typer.Option(
```
Comment (davorrunje) [inline fastkafka/_cli.py:36]: make it mandatory

**Review by davorrunje — APPROVED** [review]

File: fastkafka/_application/app.py (hunk 4 above)
Comment (kumaranvpl) [reply fastkafka/_application/app.py]: Done

File: fastkafka/_cli.py (hunk 6 above)
Comment (kumaranvpl) [reply fastkafka/_cli.py]: Done

File: fastkafka/_cli.py (hunk 7 above)
Comment (kumaranvpl) [reply fastkafka/_cli.py]: Done

File: fastkafka/_application/app.py (hunk 2 above)
Comment (kumaranvpl) [reply fastkafka/_application/app.py]: Done

File: fastkafka/_application/app.py [hunk 8]
```diff
@@ -713,7 +719,20 @@ def __init__(
...
+    def set_kafka_broker(self, kafka_broker_name: str) -> None:
+        if kafka_broker_name not in self._kafka_brokers.brokers:
+            raise ValueError(
+                f"Given kafka_broker_name '{kafka_broker_name}' is not found in kafka_brokers, available options are {self._kafka_brokers.brokers.keys()}"
+            )
+
+        broker_to_use = self._kafka_brokers.brokers[kafka_broker_name]
+        bootstrap_servers = f"{broker_to_use.url}:{broker_to_use.port}"
+        logger.info(
+            f"set_kafka_broker() : Setting bootstrap_servers value to '{bootstrap_servers}'"
```
[the remark this answers is not in the document]
Comment (kumaranvpl) [reply fastkafka/_application/app.py:734]: Done

**Review by davorrunje — APPROVED** [review]

**Pull request merged**